tools/SourceKit/tools/sourcekitd-test/sourcekitd-test.cpp

```diff
@@ -1102,6 +1105,14 @@ static int handleTestInvocation(TestOptions Opts, TestOptions &InitOpts) {
         sourcekitd_request_array_set_string(Args, SOURCEKITD_ARRAY_APPEND, Opts.ModuleCachePath.c_str());
       }
     }
+
+    if (!Opts.ResourceDir.empty()) {
+      sourcekitd_request_array_set_string(Args, SOURCEKITD_ARRAY_APPEND,
```

Comment:
@benlangmuir my only question about this PR is if I am doing something wrong here if compilerArgsAreClang is set to true. Maybe the right thing is to just not append anything in that case? What do you think?

Reply:
Interesting, I hadn't thought about this case.  We need to communicate the resource-dir here somehow, because internally we're going to create a Swift invocation that needs it, but this is expecting clang arguments and I don't think we convert clang's resource-dir to a swift one (and it's not clear that would be correct anyway).  @akyrtzi any ideas? Do we need a different way to pass this?

@gottesmm FYI the lack of resource-dir for the clang case is probably why `SourceKit/InterfaceGen/gen_header.swift` is failing.